onnxruntime/core/providers/cpu/math/gemm.cc

```diff
@@ -70,7 +70,7 @@ ONNX_CPU_OPERATOR_TYPED_KERNEL(
     KernelDefBuilder().TypeConstraint("T", DataTypeImpl::GetTensorType<double>()),
     Gemm<double>);
 
-bool GemmPackBFp32(const OpKernelInfo& info,
+bool GemmPackBFp32(AllocatorPtr alloc,
```

Comment:
This is a shared_ptr and we can pass it by reference as it'll incur an unnecessary increment in the ref count. See https://herbsutter.com/2013/06/05/gotw-91-solution-smart-pointer-parameters/

Reply:
sure- in places where we are passing the shared_ptr to other methods from within PrePack() I made them to be passed by reference to avoid ref counting, but prepack itself still takes an allocator shared_ptr by value (and triggering ref count increase). I think this is safer that we tie the life of any allocator to the life cycle of Prepack atleast.